python/lsst/daf/butler/formatters/fileFormatter.py

```diff
@@ -178,6 +248,28 @@ def write(self, inMemoryDataset, fileDescriptor):
 
         return fileDescriptor.location.pathInStore
 
+    def toBytes(self, inMemoryDataset, fileDescriptor):
+        """Serialize the Dataset to bytes based on formatter.
+
+        Parameters
+        ----------
+        inMemoryDataset : `object`
+            The Python object to serialize.
+        fileDescriptor : `FileDescriptor`
+            Identifies read type and parameters to be used for reading.
+
+        Returns
+        -------
+        serializedDataset : `str`
```

Comment:
If we are calling this `toBytes` this has to return bytes and not `str`. We are using bytes because that allows this all to work for FITS files and the like.

Reply:
I've went through all the formatters and added `encode` and `decode` where necessary and made sure the returned string starts with a `b` manually. Hopefully this is the last of the formatters mistakes (which could've probably been a separate PR).